cmd/query.go

```diff
@@ -156,6 +158,47 @@ func queryChannel(ctx *config.Context) *cobra.Command {
 	return heightFlag(cmd)
 }
 
+func queryChannelUpgrade(ctx *config.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "channel-upgrade [path-name] [chain-id]",
+		Short: "Query the channel upgrade state for the given channel id",
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			chains, _, _, err := ctx.Config.ChainsFromPath(args[0])
+			if err != nil {
+				return err
+			}
+			c := chains[args[1]]
+
+			height, err := cmd.Flags().GetUint64(flags.FlagHeight)
+			if err != nil {
+				return err
+			}
+			latestHeight, err := c.LatestHeight()
+			if err != nil {
+				return err
+			}
+			queryHeight := clienttypes.NewHeight(latestHeight.GetRevisionNumber(), uint64(height))
+			res, err := c.QueryChannelUpgrade(core.NewQueryContext(context.TODO(), queryHeight))
+			if err != nil {
+				return err
+			} else if res == nil {
+				return fmt.Errorf("failed to query for channel upgrade")
+			}
+
+			marshaler := jsonpb.Marshaler{}
+			if json, err := marshaler.MarshalToString(&res.Upgrade); err != nil {
+				fmt.Println(res.Upgrade.String())
```

Comment:
nit: Isn't it a little complicated to determine the format at runtime? I think we could also print the JSON data containing the error content instead here.

Reply:
The same applies to other commands, and it seems this should be out of the scope of this PR.

Reply:
Or can we just return `err` without displaying anything?

Reply:
@bluele I fixed this by replacing `fmt.Println` with `return err`: https://github.com/hyperledger-labs/yui-relayer/pull/143/commits/2fcd20b846c505c9c3cc03e270a56de08eafc044